Dockerfile

```diff
@@ -6,9 +6,11 @@ COPY --from=thrift /usr/local/bin/thrift /usr/local/bin/thrift
 
 WORKDIR /src
 
-COPY requirements*.txt ./
-RUN pip install -r requirements.txt
+RUN python -m venv /tmp/poetry && \
+    /tmp/poetry/bin/pip install poetry==1.8.2 && \
+    ln -s /tmp/poetry/bin/poetry /usr/local/bin/poetry
```

Comment:
I'm pretty wary about the build image pattern for virtualenvs since they aren't really portable between systems. If there's ever a mismatch between image versions or installed packages in the images, then it's not guaranteed to work.